src/main/java/io/confluent/connect/jdbc/dialect/SqlServerDatabaseDialect.java

```diff
@@ -305,6 +338,53 @@ public String buildUpsertQueryStatement(
     return builder.toString();
   }
 
+  /**
+   * If Sql Server is 2016 or newer, and time stamp mode configured against a datetime column
+   * kill task.
+   * Datetime as a Timestamp column is not supported for these versions because a Datetime casting
+   * error causes our connector to loop on the most recent record. The error arises because JDBC
+   * handles all timestamp columns as {@link java.sql.Time} and by extension to a greater precision
+   * than supported by Datetime. Since Datetime is only accurate to 3.33 MS it casts itself
+   * a higher precision recursively (3.333333 MS). However JDBC casts to higher precision
+   * non recursively (3.330000 MS). This disparity causes looping.
+   * Older MSSQL Server instances do not have this problem because it casts non-recursively.
+   * References: http://www.dbdelta.com/sql-server-2016-and-azure-sql-database-v12-breaking-change/
+   *
+   * @param rsMetadata          the result set metadata; may not be null
+   * @param timestampColumns    the timestamp columns configured; may not be null
+   * @throws ConnectException   if column type not compatible with connector
+   *                            or if there is an error accessing the result set metadata
+   */
+  @Override
+  public void validateSpecificColumnsTypes(
+          ResultSetMetaData rsMetadata,
+          List<ColumnId> timestampColumns
+  ) throws ConnectException {
+    if (verifiedSqlServerTimestamp) {
+      return;
+    }
+
+    if (versionWithBreakingDatetimeChange()) {
+      try {
+        for (int i = 0; i < rsMetadata.getColumnCount(); i++) {
+          if (rsMetadata.getColumnTypeName(i + 1).equals(DATETIME)) {
```

Comment:
why i+1? Maybe we can make the for loop better, and leave a comment?

Reply:
The indexing for Columns in rsMetadata is [indexed at 1 ](https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSetMetaData.html#getColumnName(int))(instead of 0). 
I'll leave a comment